marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -313,103 +338,98 @@ rewind = H.property $ do
       expected = filter isBeforeRollback events
   equalSet expected (MintBurn.fromRows queryResult)
 
--- See Note [cardano-testnet update] on why the code was commented out.
-
 {- | Start testnet, start mint/burn indexer on it, create a single
  mint event, put it in a transaction and submit it, find the
  generated event passed back through the indexer.
 -}
+endToEnd :: Property
+endToEnd = H.withShrinks 0 $ H.propertyOnce $ (liftIO Help.setDarwinTmpdir >>) $ H.runFinallies $ H.workspace "." $ \tempPath -> do
+  (localNodeConnectInfo, networkId, socketPath) <- Help.startTestnet tempPath
+
+  -- This is the channel we wait on to know if the event has been indexed
+  indexerChan <- liftIO IO.newChan
+  -- Start indexer
+  liftIO $ do
+    coordinator <- M.initialCoordinator 1 0
+    ch <- IO.atomically . IO.dupTChan $ M._channel coordinator
+    (loop, _indexerMVar) <-
+      M.mintBurnWorker_ 123 (IO.writeChan indexerChan) Nothing coordinator ch (tempPath </> "db.db")
+    void $ IO.async loop
+    -- Receive ChainSyncEvents and pass them on to indexer's channel
+    void $ IO.async $ do
+      let chainPoint = C.ChainPointAtGenesis :: C.ChainPoint
+      c <- defaultConfigStdout
+      withTrace c "marconi" $ \trace ->
+        let indexerWorker = withChainSyncBlockEventStream socketPath networkId [chainPoint] $
+              S.mapM_ $
+                \chainSyncEvent ->
+                  IO.atomically $ IO.writeTChan ch chainSyncEvent
+            handleException NoIntersectionFound =
+              logError trace $
+                renderStrict $
+                  layoutPretty defaultLayoutOptions $
+                    "No intersection found for chain point" <+> pretty chainPoint <> "."
+         in indexerWorker `catch` handleException :: IO ()
+
+  let address :: C.Address C.ShelleyAddr
```

Comment:
Not a request for this PR, but is there support in the node emulator to balance unbalanced tx bodies? I would envision something a follows for the test: 

I want to submit a transaction which mints some token. Now just balanced , sign and submit it. I feel like this approach would greatly simplify the test.

Reply:
It does, but not over a socket!

Reply:
Can't we use the balancing functions outside the emulator monad? If not, would it be feasible?

Reply:
No, not the emulator one, because it uses information from the wallets. But it is not so bad right now I think? It uses the balancing function from cardano-api.